plugins/pyembed/pyembed.cpp

```diff
@@ -19,11 +19,22 @@
 
 #ifdef _WIN32
 // There's an issue with Python redefining ssize_t resulting in errors - hide their definition
-#define ssize_t python_ssize_t
-#include "Python.h"
-#undef ssize_t
+  #define ssize_t python_ssize_t
+  #include "Python.h"
+  #undef ssize_t
 #else
-#include "Python.h"
+  // This Python version Python.h sets _POSIX_C_SOURCE and _XOPEN_SOURCE so we
+  // need to undefine them to prevent redefinition warnings
```

Comment:
Is it just a warning you are trying to eliminate ?  Or does the code need the newer values ?
If its just a warning then does it run ok without this ?
Would a pragma to disable warning be another way ?

Reply:
@xwang2713 has alluded to there being issues with some other third party libraries.  In my builds on centos 6 using python2.7 I've only run into warnings.  But that isn't to say that it isn't an issue on other platforms/library combinations.

Reply:
ok, so OS defines _POSIX_C_SOURCE as 200809 and pyconfig-64.h redefines it as 200112
and OS defines _XOPEN_SOURCE as 700 abd pyconfig-64.h redefines it as 600
I wish pyconfig-64.h would honor these if they were already set to something >= what it wants.

Reply:
@Michael-Gardner should the include "Python.h" happen before platform.h ?
Then platform.h undefs and defines so its ok.
Some Python documentation states that Python,h should be included before any system includes